backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java

```diff
@@ -361,15 +361,15 @@ public boolean deleteProperty(IdentificationType id, String key) {
   }
 
   @Override
-  public void addOrUpdateKeyValueProperties(AddKeyValuePropertiesRequest request) {
+  public void addKeyValueProperties(AddKeyValuePropertiesRequest request) {
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
       Transaction transaction = session.beginTransaction();
       for (KeyValueStringProperty keyValue : request.getKeyValuePropertyList()) {
         KeyValuePropertyMappingEntity.KeyValuePropertyMappingId id0 =
             KeyValuePropertyMappingEntity.createId(
                 request.getId(), keyValue.getKey(), request.getPropertyName());
         KeyValuePropertyMappingEntity existingEntity =
-            session.get(KeyValuePropertyMappingEntity.class, id0, LockMode.PESSIMISTIC_WRITE);
+            session.get(KeyValuePropertyMappingEntity.class, id0);
```

Comment:
Since this is an add the properties do not yet exist.  Any attempt to acquire a lock is guaranteed to fail.